tikv-client/src/main/java/com/pingcap/tikv/types/AbstractDateTimeType.java

```diff
@@ -39,7 +38,18 @@ ExtendedDateTime decodeDateTime(int flag, CodecDataInput cdi) {
       throw new InvalidCodecFormatException(
           "Invalid Flag type for " + getClass().getSimpleName() + ": " + flag);
     }
-    return extendedDateTime;
+
+    // Even though null is filtered out but data like 0000-00-00 exists
+    // according to MySQL JDBC behavior, it can chose the **ROUND** behavior converted to the
+    // nearest
+    // value which is 0001-01-01.
```

Comment:
are you sure it is correct? I think this logic only works for DateType rather than AbstractDateTimeType.

Reply:
If it is AbstractDateTimeType, it will be 0001-01-01 00:00:00.

Reply:
will timestamp and datetime type also meet this problem?

Reply:
You mean zero datetime? Yes.